tests/k8s/test_k8s.py

```diff
@@ -590,6 +603,86 @@ async def test_flow_with_configmap(k8s_flow_configmap, docker_images, tmpdir, lo
         raise exc
 
 
+@pytest.mark.timeout(3600)
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    'docker_images', [['test-executor', 'jinaai/jina']], indirect=True
+)
+async def test_flow_with_env_from_secret(
+    k8s_flow_env_from_secret, docker_images, tmpdir, logger
+):
+    from kubernetes import client
+
+    namespace = f'test-flow-with-env-from-secret'.lower()
+    api_client = client.ApiClient()
+    core_client = client.CoreV1Api(api_client=api_client)
+    app_client = client.AppsV1Api(api_client=api_client)
+
+    try:
+        dump_path = os.path.join('./', 'test-flow-with-env-from-secret')
+        k8s_flow_env_from_secret.to_kubernetes_yaml(dump_path, k8s_namespace=namespace)
+
+        # create namespace
+        core_client.create_namespace(client.V1Namespace(metadata=client.V1ObjectMeta(name=namespace)))
+
+        # create secret
+        secret = client.V1Secret(
+            api_version='v1',
+            kind='Secret',
+            metadata=client.V1ObjectMeta(name='mysecret'),
+            string_data={'username': 'jina', 'password': '123456'}
+        )
+        core_client.create_namespaced_secret(namespace=namespace, body=secret)
+
+        await create_all_flow_deployments_and_wait_ready(
+            dump_path,
+            namespace=namespace,
+            api_client=api_client,
+            app_client=app_client,
+            core_client=core_client,
+            deployment_replicas_expected={
+                'gateway': 1,
+                'test-executor': 1,
+            },
+            logger=logger,
+        )
+
+        # get pod name
+        executor_pod_name = core_client.list_namespaced_pod(
+            namespace=namespace,
+            label_selector=f'app=test-executor',
+        ).items[0].metadata.name
+
+        # get username and password from pod env
+        from kubernetes.stream import stream
+        username = stream(core_client.connect_get_namespaced_pod_exec,
+            executor_pod_name,
+            namespace,
+            command=['sh', '-c', 'echo $SECRET_USERNAME'],
+            stderr=True,
+            stdin=False,
+            stdout=True,
+            tty=False,
+        ).strip("\n")
+
+        password = stream(core_client.connect_get_namespaced_pod_exec,
+            executor_pod_name,
+            namespace,
+            command=['sh', '-c', 'echo $SECRET_PASSWORD'],
+            stderr=True,
+            stdin=False,
+            stdout=True,
+            tty=False,
+        ).strip("\n")
+
+        assert username == 'jina'
+        assert password == '123456'
```

Comment:
maybe also one more assert where we send a request to the executor and it returns the value of those variables ? just add an extra endpoint to test-executor